lib/channel-postmessage/src/index.ts

```diff
@@ -89,25 +105,56 @@ export class PostmsgTransport {
     });
   }
 
-  private getWindow(): Window {
+  private getFrames(target?: string): Window[] {
     if (this.config.page === 'manager') {
-      // FIXME this is a really bad idea! use a better way to do this.
-      // This finds the storybook preview iframe to send messages to.
-      const iframe = document.getElementById('storybook-preview-iframe');
-      if (!iframe) {
-        return null;
-      }
-      return iframe.contentWindow;
+      const nodes: HTMLIFrameElement[] = [
+        ...document.querySelectorAll('#storybook-preview-wrapper iframe'),
+      ];
+
+      const list = nodes
+        .filter(e => {
+          try {
+            return !!e.contentWindow && e.dataset.isStorybook !== undefined && e.id === target;
+          } catch (er) {
+            return false;
+          }
+        })
+        .map(e => e.contentWindow);
+
+      return list.length ? list : this.getCurrentFrame();
     }
-    return window.parent;
+    if (window && window.parent) {
+      return [window.parent];
+    }
+
+    return [];
+  }
+
+  private getCurrentFrame(): Window[] {
+    if (this.config.page === 'manager') {
+      const list: HTMLIFrameElement[] = [
+        ...document.querySelectorAll('#storybook-preview-wrapper span + iframe'),
```

Comment:
I don't like this implementation detail from the `Preview` component leaking into the postmessage.

Reply:
agreed, will improve